torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -96,7 +96,11 @@ Message RequestCallbackImpl::processRpc(
 
       ownerRRef->setValue(
           PythonRpcHandler::getInstance().runPythonUDF(prc.serializedPyObj()));
-      ctx.addForkOfOwner(rrefId, forkId);
+
+      if (rrefId != forkId) {
+        // This is not owner calling to self
```

Comment:
Can you add a more specific comment here on where we create a fork where rrefId == forkId (namely , when calling "remote" to self).